docker/dev/Dockerfile

```diff
@@ -158,8 +165,11 @@ USER ubuntu
 # Install Parflow
 # -----------------------------------------------------------------------------
 
-COPY --chown=ubuntu:ubuntu . $BASE_PATH/parflow/src
+# Do the pip installs on the earliest layer for faster iterative builds
```

Comment:
I've moved some pip installs earlier in the dockerfile to allow a faster incremental build of the container